statistics/handle/update.go

```diff
@@ -716,9 +719,24 @@ func (h *Handle) HandleAutoAnalyze(is infoschema.InfoSchema) (analyzed bool) {
 		rd.Shuffle(len(tbls), func(i, j int) {
 			tbls[i], tbls[j] = tbls[j], tbls[i]
 		})
+
+		tids := make([]int64, 0, len(tbls))
+		for _, tbl := range tbls {
+			tids = append(tids, tbl.Meta().ID)
+		}
+
+		checkedTableIDs, err := h.QueryTablesLockedStatuses(tids...)
+		if err != nil {
+			logutil.BgLogger().Error("check table lock failed",
+				zap.String("category", "stats"), zap.Error(err))
+			continue
+		}
+
 		for _, tbl := range tbls {
-			//if table locked, skip analyze
-			if h.IsTableLocked(tbl.Meta().ID) {
+			// If table locked, skip analyze.
+			if checkedTableIDs[tbl.Meta().ID] {
+				logutil.BgLogger().Info("skip analyze locked table", zap.String("category", "stats"),
+					zap.String("db", db), zap.String("table", tbl.Meta().Name.O))
```

Comment:
Please add some comments here to notice this check is not complete. For example, some other TiDB node may lock or unlock this table between this check and the analyze command.

Reply:
Better to enhance or remove this logic in the future.